includes/class-sensei-course.php

```diff
@@ -2090,6 +2090,18 @@ public function load_user_courses_content( $user = false ) {
 
 					}
 				}
+					/**
+					 * Publicly displays links related to the completed course
+					 *
+					 * @since 4.16.1
+					 * @hook sensei_results_completed_links
+					 *
+					 * @param {int} $course_id The ID of the course.
+					 * @param {int} $user_id The user ID of the learner.
+					 *
+					 * @return {string} HTML output of the links.
+					 */
+					$complete_html .= apply_filters( 'sensei_results_completed_links', '', $course_item->ID, $user->ID );
```

Comment:
Instead of introducing a new filter, do you think we can reuse the old one? I was thinking that we might refactor a bit to make the hook a bit more accessible. Let me know what you think of the following patch:

```patch
diff --git a/includes/class-sensei-course.php b/includes/class-sensei-course.php
index 5df6d9d9a..67d4d6388 100755
--- a/includes/class-sensei-course.php
+++ b/includes/class-sensei-course.php
@@ -2067,41 +2067,23 @@ class Sensei_Course {

                                                $complete_html .= $this->get_progress_meter( 100 );

-                               if ( $manage ) {
-                                       $has_quizzes = Sensei()->course->course_quizzes( $course_item->ID, true );
-
-                                       // Output only if there is content to display
-                                       if ( self::has_results_links( $course_item->ID ) || $has_quizzes ) {
-                                               $complete_html .= '<p class="sensei-results-links">';
-                                               $results_link   = '';
-
-                                               if ( $has_quizzes ) {
-                                                       $results_link = '<a class="button view-results" href="'
-                                                               . esc_url( self::get_view_results_link( $course_item->ID ) )
-                                                               . '">' . esc_html__( 'View Results', 'sensei-lms' )
-                                                               . '</a>';
-                                               }
-
-                                               /**
-                                                * Filter documented in Sensei_Course::the_course_action_buttons
-                                                */
-                                               $complete_html .= apply_filters( 'sensei_results_links', $results_link, $course_item->ID );
-                                               $complete_html .= '</p>';
+                               $results_link = '';
+                               if ( $manage && Sensei()->course->course_quizzes( $course_item->ID, true ) ) {
+                                       $results_link = '<a class="button view-results" href="'
+                                               . esc_url( self::get_view_results_link( $course_item->ID ) )
+                                               . '">' . esc_html__( 'View Results', 'sensei-lms' )
+                                               . '</a>';
+                               }

-                                       }
+                               /**
+                                * Filter documented in Sensei_Course::the_course_action_buttons
+                                */
+                               $results_links = apply_filters( 'sensei_results_links', $results_link, $course_item->ID );
+                               if ( $results_links ) {
+                                       $complete_html .= '<p class="sensei-results-links">';
+                                       $complete_html .= $results_links;
+                                       $complete_html .= '</p>';
                                }
-                                       /**
-                                        * Publicly displays links related to the completed course
-                                        *
-                                        * @since 4.16.1
-                                        * @hook sensei_results_completed_links
-                                        *
-                                        * @param {int} $course_id The ID of the course.
-                                        * @param {int} $user_id The user ID of the learner.
-                                        *
-                                        * @return {string} HTML output of the links.
-                                        */
-                                       $complete_html .= apply_filters( 'sensei_results_completed_links', '', $course_item->ID, $user->ID );

                                        $complete_html .= '</section>';
```

This should preserve the old logic and make it easier for third parties to hook in. I don't think we need the `self::has_results_links` check in this case, so I've removed it from the logic.

LMKWYT.

Reply:
Thanks @m1r0 ! 

I'm all for this if it won't have any unintended side effects. I added a new filter thinking having it applied inside the `$manage` was intentional but if it isn't, this makes sense.

If that filter isn't supposed to be suppressed like that, your patch makes perfect sense!

Reply:
Updated in 468d86548

Reply:
@m1r0 I had to make one update to apply the leaner's user ID to the `apply_filter()` statement here. That enables us to determine if a user has access to view the certificate. I made the change in 03a4382